app/controllers/templates_controller.rb

```diff
@@ -79,15 +79,21 @@ def auto_complete_controller_name
   end
 
   def preview
-    base = @template.preview_host_collection
-    @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    if params[:id]
+      find_resource
+      base = @template.preview_host_collection
+      @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    else
+      type = request.original_url.split("/")[4]
+      @host = Host.find(params[:preview_host_id])
+      @template = resource_class.new(params[type])
+    end
     if @host.nil?
       render :text => _('No host could be found for rendering the template'), :status => :not_found
       return
     end
-
     @template.template = params[:template]
-    safe_render(@template)
+    safe_render(params[:id] ? @template : @template.delete)
```

Comment:
@alongoldboim The record is not in the db if the user just previews and closes the window without saving, it would be just in memory then removed on the next pass of the garbage collector.